### ISSUE TITLE
Umlauts/UTF-8 causes wrong error column to be shown

### ISSUE THREAD

**Issue by hoxu** [dense-analysis/ale#3394]
## Information

**VIM version**

```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Oct  7 2020 19:37:01)
macOS version
```

Operating System: MacOS

## What went wrong

Umlauts (åäö) / UTF-8 causes wrong error column to be shown visually.

## Reproducing the bug

Write a line containing umlauts/UTF-8 and a syntax error. Lint with `clj-kondo`, which displays character, not byte, column. Observe error column being shown left from the correct position.

### :ALEInfo

```
 Current Filetype: clojure
Available Linters: ['clj-kondo', 'joker']
  Enabled Linters: ['clj-kondo']
  Ignored Linters: []
 Suggested Fixers: 
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'clojure': ['clj-kondo']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = '>>'
let g:ale_sign_info = '--'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = '>>'
let g:ale_sign_style_warning = '--'
let g:ale_sign_warning = '--'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(finished - exit code 3) ['/bin/bash', '-c', 'clj-kondo --cache --lint ''/var/folders/tl/9_nnmfjs0lj4g924c1qwxdjh0000gp/T/vPEjYtY/22/xxx.clj''']

<<<OUTPUT STARTS>>>
/var/folders/tl/9_nnmfjs0lj4g924c1qwxdjh0000gp/T/vPEjYtY/22/xxx.clj:15:80: error: unresolved symbol data
linting took 20ms, errors: 1, warnings: 0
<<<OUTPUT ENDS>>>
```

**Comment by nospam2998** [comment]
Please have a look at `:help ale-api`, specifically at the documentation for `vcol`, quoted below for convenience:

> `vcol` - Defaults to `0`.
>
> If set to `1`, ALE will convert virtual column positions for `col` and `end_col` to byte column positions. If the buffer is changed in-between checking it and displaying the results, the calculated byte column positions will probably be wrong.

My paraphrased understanding of the above is that it convert character offsets to byte offsets.

I would thus not be surprised if this would be solved merely by adding this variable to `HandleCljKondoFormat()` and updating the tests. A few single line additions. Please feel free to see #2096 for patch inspiration.

Pull requests are welcome, and are best written by those having easy access to the linter in question.
